File: backend/model/gift_purchase_order.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// GiftPurchaseOrder 用心意（余额）购买礼物进入背包的订单记录
type GiftPurchaseOrder struct {
	ID          uint           `gorm:"primarykey" json:"id"`
	UserID      uint           `gorm:"not null;index" json:"user_id"`
	OrderNo     string         `gorm:"size:64;uniqueIndex;not null" json:"order_no"`
	GiftID      uint           `gorm:"not null;index" json:"gift_id"`
	GiftName    string         `gorm:"size:128;not null" json:"gift_name"`
	Quantity    int            `gorm:"not null" json:"quantity"`
	UnitPrice   float64        `gorm:"not null" json:"unit_price"`
	TotalAmount float64        `gorm:"not null" json:"total_amount"`
	PayMethod   string         `gorm:"size:20;not null;default:wallet" json:"pay_method"`
	Status      string         `gorm:"size:20;not null;default:paid" json:"status"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

File: lib/models/gift_purchase_order.dart
class GiftPurchaseOrder {
  final String id;
  final String userId;
  final String orderNo;
  final String giftId;
  final String giftName;
  final int quantity;
  final double unitPrice;
  final double totalAmount;
  final String payMethod;
  final String status;
  final String createdAt;

  GiftPurchaseOrder({
    required this.id,
    required this.userId,
    required this.orderNo,
    required this.giftId,
    required this.giftName,
    required this.quantity,
    required this.unitPrice,
    required this.totalAmount,
    required this.payMethod,
    required this.status,
    required this.createdAt,
  });

  factory GiftPurchaseOrder.fromJson(Map<String, dynamic> json) {
    return GiftPurchaseOrder(
      id: json['id']?.toString() ?? '',
      userId: json['user_id']?.toString() ?? '',
      orderNo: json['order_no'] as String? ?? '',
      giftId: json['gift_id']?.toString() ?? '',
      giftName: json['gift_name'] as String? ?? '',
      quantity: (json['quantity'] as num?)?.toInt() ?? 0,
      unitPrice: (json['unit_price'] as num?)?.toDouble() ?? 0.0,
      totalAmount: (json['total_amount'] as num?)?.toDouble() ?? 0.0,
      payMethod: json['pay_method'] as String? ?? '',
      status: json['status'] as String? ?? '',
      createdAt: json['created_at'] as String? ?? '',
    );
  }
}
